plugins/module_utils/common_api.py

```diff
@@ -0,0 +1,582 @@
+
+
+def is_using_tls(auth_data):
+    return auth_data['tls_verify'] or auth_data['tls']
+
+
+def get_connect_params(auth_data, fail_function):
+    if is_using_tls(auth_data):
+        auth['docker_host'] = auth_data['docker_host'].replace('tcp://', 'https://')
+
+    result = dict(
+        base_url=auth_data['docker_host'],
+        version=auth_data['api_version'],
+        timeout=auth_data['timeout'],
+    )
+
+    if auth_data['tls_verify']:
+        # TLS with verification
+        tls_config = dict(
+            verify=True,
+            assert_hostname=auth_data['tls_hostname'],
+            ssl_version=auth_data['ssl_version'],
+            fail_function=fail_function,
+        )
+        if auth_data['cert_path'] and auth_data['key_path']:
+            tls_config['client_cert'] = (auth_data['cert_path'], auth_data['key_path'])
+        if auth_data['cacert_path']:
+            tls_config['ca_cert'] = auth_data['cacert_path']
+        result['tls'] = _get_tls_config(**tls_config)
+    elif auth_data['tls']:
+        # TLS without verification
+        tls_config = dict(
+            verify=False,
+            ssl_version=auth_data['ssl_version'],
+            fail_function=fail_function,
+        )
+        if auth_data['cert_path'] and auth_data['key_path']:
+            tls_config['client_cert'] = (auth_data['cert_path'], auth_data['key_path'])
+        result['tls'] = _get_tls_config(**tls_config)
+
+    if auth_data.get('use_ssh_client'):
+        result['use_ssh_client'] = True
+
+    # No TLS
+    return result
+
+
+class AnsibleDockerClientBase(Client):
+    def __init__(self, min_docker_api_version=None):
+        self._connect_params = get_connect_params(self.auth_params, fail_function=self.fail)
+
+        try:
+            super(AnsibleDockerClientBase, self).__init__(**self._connect_params)
+            self.docker_api_version_str = self.api_version
+        except MissingRequirementException as exc:
+            self.fail(missing_required_lib(exc.requirement), exception=exc.import_exception)
+        except APIError as exc:
+            self.fail("Docker API error: %s" % exc)
+        except Exception as exc:
+            self.fail("Error connecting: %s" % exc)
+
+        self.docker_api_version = LooseVersion(self.docker_api_version_str)
+        if min_docker_api_version is not None:
+            if self.docker_api_version < LooseVersion(min_docker_api_version):
+                self.fail('Docker API version is %s. Minimum version required is %s.' % (self.docker_api_version_str, min_docker_api_version))
+
+    def log(self, msg, pretty_print=False):
+        pass
+        # if self.debug:
+        #     log_file = open('docker.log', 'a')
+        #     if pretty_print:
+        #         log_file.write(json.dumps(msg, sort_keys=True, indent=4, separators=(',', ': ')))
+        #         log_file.write(u'\n')
+        #     else:
+        #         log_file.write(msg + u'\n')
```

Comment:
Is this meant to stay in?

Reply:
This is copied from common.py, having it commented out makes it easy to comment it back in and have some basic logging on `debug=true`. So as long as it looks the same in common.py it should also be in common_api.py :)